itests/hive-unit-hadoop2/src/test/java/org/apache/hadoop/hive/metastore/security/TestHadoopAuthBridge23.java

```diff
@@ -204,13 +204,22 @@ public void testDelegationTokenSharedStore() throws Exception {
       // expected
     }
 
-    // token expiration
+    // token Renewal
     MyTokenStore.TOKEN_STORE.addToken(d,
         new DelegationTokenInformation(0, t.getPassword()));
     Assert.assertNotNull(MyTokenStore.TOKEN_STORE.getToken(d));
     anotherManager.removeExpiredTokens();
+    Assert.assertTrue(MyTokenStore.TOKEN_STORE.getToken(d).getRenewDate() > 0);
```

Comment:
It may throw NPE because maxtime was not set.

Reply:
Its not about accessing maxtime, if maxtime is zero above statement "removeExpiredTokens()" will remove this token. After token removed calling getRenewDate() will throw NPE

Reply:
No it will not give NPE here as maxDate is not zero here. In this case it is one hour actually.